### PR TITLE
8330534: Update nsk/jdwp tests to use driver instead of othervm

### PR DESCRIPTION
**Pull request by lmesnik**
The jdwp tests use debugger and debugee. There is no goal to execute debugger part with all VM flags, they are needed to be used with debugee VM only.
The change is all tests is to don't use System.exit() and use 'driver' instead of othervm.
test/hotspot/jtreg/vmTestbase/nsk/share/jpda/DebugeeBinder.java
is updated to correctly set classpath for debugee

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8330534](https://bugs.openjdk.org/browse/JDK-8330534): Update nsk/jdwp tests to use driver instead of othervm (**Sub-task** - P4)

### Reviewers
 * [Serguei Spitsyn](https://openjdk.org/census#sspitsyn) (@sspitsyn - **Reviewer**) ⚠️ Review applies to [122446b6](https://git.openjdk.org/jdk/pull/18826/files/122446b6884848f512d15b4be8146273e44f452f)
 * [Chris Plummer](https://openjdk.org/census#cjplummer) (@plummercj - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/18826/head:pull/18826` \
`$ git checkout pull/18826`

Update a local copy of the PR: \
`$ git checkout pull/18826` \
`$ git pull https://git.openjdk.org/jdk.git pull/18826/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 18826`

View PR using the GUI difftool: \
`$ git pr show -t 18826`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/18826.diff">https://git.openjdk.org/jdk/pull/18826.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/18826#issuecomment-2062165042)

### CODE REVIEW

File: test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/ForceEarlyReturn/forceEarlyReturn001/forceEarlyReturn001.java [hunk 1]
```diff
@@ -123,8 +123,11 @@ protected String getDebugeeClassName() {
         return "nsk.jdwp.ThreadReference.ForceEarlyReturn.forceEarlyReturn001.forceEarlyReturn001a";
     }
 
-    public static void main(String[] argv) {
-        System.exit(run(argv, System.out) + Consts.JCK_STATUS_BASE);
+    public static void main (String argv[]) {
+        int result = run(argv,System.out);
```
Comment (turbanoff) [inline test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/ForceEarlyReturn/forceEarlyReturn001/forceEarlyReturn001.java:127]:
```suggestion
        int result = run(argv, System.out);
```

File: test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/Interrupt/interrupt001.java [hunk 2]
```diff
@@ -90,7 +90,10 @@ public class interrupt001 {
      * Start test from command line.
      */
     public static void main (String argv[]) {
-        System.exit(run(argv,System.out) + JCK_STATUS_BASE);
+        int result = run(argv,System.out);
```
Comment (turbanoff) [inline test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/Interrupt/interrupt001.java:93]:
```suggestion
        int result = run(argv, System.out);
```

File: test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/Name/name001.java [hunk 3]
```diff
@@ -87,7 +87,10 @@ public class name001 {
      * Start test from command line.
      */
     public static void main (String argv[]) {
-        System.exit(run(argv,System.out) + JCK_STATUS_BASE);
+        int result = run(argv,System.out);
```
Comment (turbanoff) [inline test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/Name/name001.java:90]:
```suggestion
        int result = run(argv, System.out);
```

File: test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/OwnedMonitors/ownmonitors001.java [hunk 4]
```diff
@@ -98,7 +98,10 @@ public class ownmonitors001 {
      * Start test from command line.
      */
     public static void main (String argv[]) {
-        System.exit(run(argv,System.out) + JCK_STATUS_BASE);
+        int result = run(argv,System.out);
```
Comment (turbanoff) [inline test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/OwnedMonitors/ownmonitors001.java:101]:
```suggestion
        int result = run(argv, System.out);
```

File: test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/Interrupt/interrupt001.java (hunk 2 above)
Comment (sspitsyn) [reply test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/Interrupt/interrupt001.java:93]: Yes, this missing space is all over this fix. Need to fix this formatting.

**Review by sspitsyn — APPROVED** [review]
This looks okay in general.
There is a missing space in half of files in method `run()` parameters noticed by Andrei.
Also, Chris had some concerns about converting to the `driver` mode.

File: test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/OwnedMonitors/ownmonitors001.java (hunk 4 above)
Comment (lmesnik) [reply test/hotspot/jtreg/vmTestbase/nsk/jdwp/ThreadReference/OwnedMonitors/ownmonitors001.java]: Thanks, I update the indentation in all places.

**Comment by plummercj** [comment]
There is something wrong with this diff. The original commit showed a lot of diffs to change main/othervm to driver (the main purpose of this PR), but now I don't see any of those changes and instead for the most part all I see is the very minor change to add a space in the `run()` call arguments. I think for some reason only your 2nd commit is being applied to this PR and the first commit was lost. However, the loss of the first commit is not showing up in the incremental diff of the two commits. I'm not sure what's going on.

**Comment by lmesnik** [comment]
Thanks! I've updated also 'jdi' tests by this space deletion. Reverted the changes, the PR is clean now. I'll update jdi tests separately.

**Review by plummercj — APPROVED** [review]
Looks good.

**Pull request closed without merging**